### PR TITLE
CLOUD-619: Add configs for tests on GKE latest

### PR DESCRIPTION
**Pull request by egegunes**

### CODE REVIEW

File: cloud/jenkins/psmdb_operator_gke_alpha.groovy [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+    post {
+        always {
+            setTestsresults()
+            withCredentials([string(credentialsId: 'GCP_PROJECT_ID', variable: 'GCP_PROJECT'), file(credentialsId: 'gcloud-alpha-key-file', variable: 'CLIENT_SECRET_FILE')]) {
+                sh '''
+                    export CLUSTER_NAME=$(echo jenkins-psmdb-$(git -C source rev-parse --short HEAD) | tr '[:upper:]' '[:lower:]')
+                    source $HOME/google-cloud-sdk/path.bash.inc
+                    gcloud auth activate-service-account alpha-svc-acct@"${GCP_PROJECT}".iam.gserviceaccount.com --key-file=$CLIENT_SECRET_FILE
+                    gcloud config set project $GCP_PROJECT
+                    gcloud alpha container clusters delete --zone us-central1-a $CLUSTER_NAME-basic $CLUSTER_NAME-scaling $CLUSTER_NAME-selfhealing $CLUSTER_NAME-backups | true
```
Comment (hors) [inline cloud/jenkins/psmdb_operator_gke_alpha.groovy:301]: please delete the cluster in the following way https://github.com/percona/percona-xtradb-cluster-operator/blob/main/Jenkinsfile#L387

File: cloud/jenkins/psmdb_operator_gke_alpha.groovy [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+                '''
+            }
+            sh '''
+                sudo docker rmi -f \$(sudo docker images -q) || true
```
Comment (hors) [inline cloud/jenkins/psmdb_operator_gke_alpha.groovy:305]: https://github.com/percona/percona-xtradb-cluster-operator/blob/main/Jenkinsfile#L389 please check it

File: cloud/jenkins/pxc_operator_gke_alpha.groovy [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+                sh '''
+                    export CLUSTER_NAME=$(echo jenkins-pxc-$(git -C source rev-parse --short HEAD) | tr '[:upper:]' '[:lower:]')
+                    source $HOME/google-cloud-sdk/path.bash.inc
+                    gcloud auth activate-service-account alpha-svc-acct@"${GCP_PROJECT}".iam.gserviceaccount.com --key-file=$CLIENT_SECRET_FILE
+                    gcloud config set project $GCP_PROJECT
+                    gcloud alpha container clusters delete --zone us-central1-a $CLUSTER_NAME-basic $CLUSTER_NAME-scaling $CLUSTER_NAME-selfhealing $CLUSTER_NAME-backups $CLUSTER_NAME-bigdata $CLUSTER_NAME-upgrade | true
+                '''
+            }
+            sh '''
+                sudo docker rmi -f \$(sudo docker images -q) || true
```
Comment (hors) [inline cloud/jenkins/pxc_operator_gke_alpha.groovy:352]: https://github.com/percona/percona-xtradb-cluster-operator/blob/main/Jenkinsfile#L389 please check it

File: cloud/jenkins/pxc_operator_gke_alpha.groovy [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+            step([$class: 'JUnitResultArchiver', testResults: '*.xml', healthScaleFactor: 1.0])
+            archiveArtifacts '*.xml'
+
+            withCredentials([string(credentialsId: 'GCP_PROJECT_ID', variable: 'GCP_PROJECT'), file(credentialsId: 'gcloud-alpha-key-file', variable: 'CLIENT_SECRET_FILE')]) {
+                sh '''
+                    export CLUSTER_NAME=$(echo jenkins-pxc-$(git -C source rev-parse --short HEAD) | tr '[:upper:]' '[:lower:]')
+                    source $HOME/google-cloud-sdk/path.bash.inc
+                    gcloud auth activate-service-account alpha-svc-acct@"${GCP_PROJECT}".iam.gserviceaccount.com --key-file=$CLIENT_SECRET_FILE
+                    gcloud config set project $GCP_PROJECT
+                    gcloud alpha container clusters delete --zone us-central1-a $CLUSTER_NAME-basic $CLUSTER_NAME-scaling $CLUSTER_NAME-selfhealing $CLUSTER_NAME-backups $CLUSTER_NAME-bigdata $CLUSTER_NAME-upgrade | true
```
Comment (hors) [inline cloud/jenkins/pxc_operator_gke_alpha.groovy:348]: please delete the cluster in the following way https://github.com/percona/percona-xtradb-cluster-operator/blob/main/Jenkinsfile#L387

**Review by hors — CHANGES_REQUESTED** [review]

**Comment by hors** [comment]
What about the latest openshift, I think we need to test it too.

File: cloud/jenkins/pgo-operator-gke-alpha.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      - timed: "0 0 * * 3"
+    pipeline-scm:
+      scm:
+        - git:
+            url: https://github.com/Percona-Lab/jenkins-pipelines.git
+            branches:
+            - master
+            wipe-workspace: false
+      lightweight-checkout: true
+      script-path: cloud/jenkins/pgo_operator_gke_version.groovy
```
[the remark this answers is not in the document]
Comment (egegunes) [reply cloud/jenkins/pgo-operator-gke-alpha.yml]: https://github.com/Percona-Lab/jenkins-pipelines/pull/1009/commits/b43c55ab86b9ef473cafc63524894ee8c5b9cebd

File: cloud/jenkins/pgo_operator_gke_alpha.groovy [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+    post {
+        always {
+            setTestsresults()
+            withCredentials([string(credentialsId: 'GCP_PROJECT_ID', variable: 'GCP_PROJECT'), file(credentialsId: 'gcloud-alpha-key-file', variable: 'CLIENT_SECRET_FILE')]) {
+                sh '''
+                    export CLUSTER_NAME=$(echo jenkins-pgo-$(git -C source rev-parse --short HEAD) | tr '[:upper:]' '[:lower:]')
+                    source $HOME/google-cloud-sdk/path.bash.inc
+                    gcloud auth activate-service-account alpha-svc-acct@"${GCP_PROJECT}".iam.gserviceaccount.com --key-file=$CLIENT_SECRET_FILE
+                    gcloud config set project $GCP_PROJECT
+                    gcloud alpha container clusters delete --zone us-central1-a $CLUSTER_NAME-sandbox || true
```
[the remark this answers is not in the document]
Comment (egegunes) [reply cloud/jenkins/pgo_operator_gke_alpha.groovy]: https://github.com/Percona-Lab/jenkins-pipelines/pull/1009/commits/b43c55ab86b9ef473cafc63524894ee8c5b9cebd

File: cloud/jenkins/pgo_operator_gke_alpha.groovy [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+                '''
+            }
+            sh '''
+                sudo docker rmi -f \$(sudo docker images -q) || true
```
[the remark this answers is not in the document]
Comment (egegunes) [reply cloud/jenkins/pgo_operator_gke_alpha.groovy]: https://github.com/Percona-Lab/jenkins-pipelines/pull/1009/commits/b43c55ab86b9ef473cafc63524894ee8c5b9cebd

File: cloud/jenkins/psmdb_operator_gke_alpha.groovy (hunk 1 above)
Comment (egegunes) [reply cloud/jenkins/psmdb_operator_gke_alpha.groovy]: https://github.com/Percona-Lab/jenkins-pipelines/pull/1009/commits/b43c55ab86b9ef473cafc63524894ee8c5b9cebd

File: cloud/jenkins/psmdb_operator_gke_alpha.groovy (hunk 2 above)
Comment (egegunes) [reply cloud/jenkins/psmdb_operator_gke_alpha.groovy]: https://github.com/Percona-Lab/jenkins-pipelines/pull/1009/commits/b43c55ab86b9ef473cafc63524894ee8c5b9cebd

File: cloud/jenkins/pxc_operator_gke_alpha.groovy (hunk 3 above)
Comment (egegunes) [reply cloud/jenkins/pxc_operator_gke_alpha.groovy]: https://github.com/Percona-Lab/jenkins-pipelines/pull/1009/commits/b43c55ab86b9ef473cafc63524894ee8c5b9cebd

File: cloud/jenkins/pxc_operator_gke_alpha.groovy (hunk 4 above)
Comment (egegunes) [reply cloud/jenkins/pxc_operator_gke_alpha.groovy]: https://github.com/Percona-Lab/jenkins-pipelines/pull/1009/commits/b43c55ab86b9ef473cafc63524894ee8c5b9cebd

**Comment by egegunes** [comment]
@hors openshift: https://github.com/Percona-Lab/jenkins-pipelines/pull/1009/commits/953520a2f0b72d8ffcef03c32b4c9e1db1e308ce

File: cloud/jenkins/pgo_operator_aws_openshift-latest.groovy [hunk 8]
```diff
@@ -0,0 +1,317 @@
+void pushArtifactFile(String FILE_NAME) {
+    echo "Push $FILE_NAME file to S3!"
```
Comment (cap1984) [inline cloud/jenkins/pgo_operator_aws_openshift-latest.groovy:2]: I believe we can put some fancy "latest" logic into existing openshift job. 
The same is true for gke.

If you have any objections please comment

Comment (egegunes) [reply cloud/jenkins/pgo_operator_aws_openshift-latest.groovy:2]: Yes, we can. But @hors said that we should duplicate the jobs

**Review by tplavcic — CHANGES_REQUESTED** [review]
Should we maybe export functions to external groovy file and just load them into every pipeline with `load` and use as a module? (note that if this makes sense it might make sense to do it after this is merged since other "non-latest" jobs have the same issue).
Otherwise looks ok to me, just make sure to test every job.

**Comment by hors** [comment]
> Should we maybe export functions to external groovy file and just load them into every pipeline with `load` and use as a module? (note that if this makes sense it might make sense to do it after this is merged since other "non-latest" jobs have the same issue).
> Otherwise looks ok to me, just make sure to test every job.

We definitely should do it but via a separate task. I will create the task.

**Review by hors — APPROVED** [review]

File: cloud/jenkins/pgo_operator_gke_alpha.groovy [hunk 9]
```diff
@@ -0,0 +1,297 @@
+void CreateCluster(String CLUSTER_PREFIX) {
```
Comment (tplavcic) [inline cloud/jenkins/pgo_operator_gke_alpha.groovy:1]: Minor thing: CLUSTER_PREFIX is actually used as cluster suffix

File: cloud/jenkins/psmdb_operator_aws_openshift-latest.groovy [hunk 10]
```diff
@@ -0,0 +1,283 @@
...
+
+    withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'AMI/OVF', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+        sh """
+            S3_PATH=s3://percona-jenkins-artifactory/\$JOB_NAME/\$(git -C source describe --always --dirty)
+            aws s3 cp --quiet \$S3_PATH/${FILE_NAME} ${FILE_NAME} || :
+        """
+    }
+}
+
+TestsReport = '<testsuite  name=\\"PSMDB\\">\n'
```
Comment (tplavcic) [inline cloud/jenkins/psmdb_operator_aws_openshift-latest.groovy:25]: double space

File: cloud/jenkins/psmdb_operator_gke_alpha.groovy [hunk 11]
```diff
@@ -0,0 +1,313 @@
+void CreateCluster(String CLUSTER_PREFIX) {
```
Comment (tplavcic) [inline cloud/jenkins/psmdb_operator_gke_alpha.groovy:1]: CLUSTER_PREFIX is used as suffix, so I propose to rename it everywhere

File: cloud/jenkins/pxc-operator-gke-alpha.yml [hunk 12]
```diff
@@ -0,0 +1,48 @@
+- job:
```
Comment (tplavcic) [inline cloud/jenkins/pxc-operator-gke-alpha.yml:1]: gke jobs are called `alpha`, but openshift `latest`- I propose for these jobs to use `latest` for all so it's consistent

File: cloud/jenkins/pgo_operator_aws_openshift-latest.groovy [hunk 13]
```diff
@@ -0,0 +1,317 @@
...
+
+    withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'AMI/OVF', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+        sh """
+            S3_PATH=s3://percona-jenkins-artifactory/\$JOB_NAME/\$(git -C source describe --always --dirty)
+            aws s3 cp --quiet \$S3_PATH/${FILE_NAME} ${FILE_NAME} || :
+        """
+    }
+}
+
+TestsReport = '<testsuite  name=\\"PGO\\">\n'
```
Comment (tplavcic) [inline cloud/jenkins/pgo_operator_aws_openshift-latest.groovy:25]: double space

File: cloud/jenkins/pxc_operator_aws_openshift-latest.groovy [hunk 14]
```diff
@@ -0,0 +1,350 @@
...
+
+    withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'AMI/OVF', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+        sh """
+            S3_PATH=s3://percona-jenkins-artifactory/\$JOB_NAME/\$(git -C source describe --always --dirty)
+            aws s3 cp --quiet \$S3_PATH/${FILE_NAME} ${FILE_NAME} || :
+        """
+    }
+}
+
+TestsReport = '<testsuite  name=\\"PXC\\">\n'
```
Comment (tplavcic) [inline cloud/jenkins/pxc_operator_aws_openshift-latest.groovy:31]: double space

File: cloud/jenkins/pxc_operator_gke_alpha.groovy [hunk 15]
```diff
@@ -0,0 +1,359 @@
...
+
+    withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'AMI/OVF', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+        sh """
+            S3_PATH=s3://percona-jenkins-artifactory/\$JOB_NAME/${env.GIT_SHORT_COMMIT}
+            aws s3 cp --quiet \$S3_PATH/${FILE_NAME} ${FILE_NAME} || :
+        """
+    }
+}
+
+TestsReport = '<testsuite  name=\\"PXC\\">\n'
```
Comment (tplavcic) [inline cloud/jenkins/pxc_operator_gke_alpha.groovy:48]: double space

**Review by tplavcic — CHANGES_REQUESTED** [review]

File: cloud/jenkins/pxc_operator_gke_alpha.groovy (hunk 15 above)
Comment (egegunes) [reply cloud/jenkins/pxc_operator_gke_alpha.groovy]: https://github.com/Percona-Lab/jenkins-pipelines/pull/1009/commits/5e753a5a785f2f85ea63895d0d7eccf57bfcd61f

File: cloud/jenkins/pxc_operator_aws_openshift-latest.groovy (hunk 14 above)
Comment (egegunes) [reply cloud/jenkins/pxc_operator_aws_openshift-latest.groovy]: https://github.com/Percona-Lab/jenkins-pipelines/pull/1009/commits/5e753a5a785f2f85ea63895d0d7eccf57bfcd61f

File: cloud/jenkins/pgo_operator_aws_openshift-latest.groovy (hunk 13 above)
Comment (egegunes) [reply cloud/jenkins/pgo_operator_aws_openshift-latest.groovy]: https://github.com/Percona-Lab/jenkins-pipelines/pull/1009/commits/5e753a5a785f2f85ea63895d0d7eccf57bfcd61f

File: cloud/jenkins/pxc-operator-gke-alpha.yml (hunk 12 above)
Comment (egegunes) [reply cloud/jenkins/pxc-operator-gke-alpha.yml]: https://github.com/Percona-Lab/jenkins-pipelines/pull/1009/commits/5e753a5a785f2f85ea63895d0d7eccf57bfcd61f

File: cloud/jenkins/psmdb_operator_gke_alpha.groovy (hunk 11 above)
Comment (egegunes) [reply cloud/jenkins/psmdb_operator_gke_alpha.groovy]: https://github.com/Percona-Lab/jenkins-pipelines/pull/1009/commits/5e753a5a785f2f85ea63895d0d7eccf57bfcd61f

File: cloud/jenkins/psmdb_operator_aws_openshift-latest.groovy (hunk 10 above)
Comment (egegunes) [reply cloud/jenkins/psmdb_operator_aws_openshift-latest.groovy]: https://github.com/Percona-Lab/jenkins-pipelines/pull/1009/commits/5e753a5a785f2f85ea63895d0d7eccf57bfcd61f

File: cloud/jenkins/pgo_operator_gke_alpha.groovy (hunk 9 above)
Comment (egegunes) [reply cloud/jenkins/pgo_operator_gke_alpha.groovy]: https://github.com/Percona-Lab/jenkins-pipelines/pull/1009/commits/5e753a5a785f2f85ea63895d0d7eccf57bfcd61f

**Review by tplavcic — APPROVED** [review]

**Pull request merged**